multimodal/src/autogluon/multimodal/utils/data.py

```diff
@@ -427,12 +429,21 @@ def data_to_df(
     elif isinstance(data, dict):
         data = pd.DataFrame(data)
     elif isinstance(data, list):
-        if header is None:
-            data = pd.DataFrame(data)
+        assert len(data) > 0, f"Expected data to have length > 0, but got {data} of len {len(data)}"
+        if contains_valid_images(data):
+            data_dict = {"image": data}
```

Comment:
Is it necessary to check if a list of image paths and use header `image`? Whether image list or not doesn't affect creating the dataframe. pandas creates a dataframe with header `0`. I don't think header `image` is required in later data processing and model forward.

Reply:
let me double check. I think I saw at some point the code actually uses the header `image`